File: app/src/main/java/com/p5m/puzzledroid/PuzzleDroidApplication.java
package com.p5m.puzzledroid;

import android.app.Application;
import android.media.MediaPlayer;

import timber.log.Timber;

/**
 * Override Application to set up some initial configuration
 */
public class PuzzleDroidApplication extends Application {
    private static PuzzleDroidApplication sInstance;
    public MediaPlayer mp;
    /**
     * Called before the first screen is shown to the user
     */
    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.DebugTree());

        //Create music and start
        sInstance = this;
        mp = MediaPlayer.create(this, R.raw.song1);
        mp.start();
    }
    public static PuzzleDroidApplication getInstance() {
        return PuzzleDroidApplication.sInstance;
    }


}

File: app/src/main/java/com/p5m/puzzledroid/PuzzledroidApplication.kt
package com.p5m.puzzledroid

import android.app.Application
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import timber.log.Timber

/**
 * Override Application to set up some initial configuration
 */
class PuzzledroidApplication: Application() {
    private val applicationScope = CoroutineScope(Dispatchers.Default)

    /**
     * Called before the first screen is shown to the user
     */
    override fun onCreate() {
        super.onCreate()
        delayedInit()
    }

    /**
     * Used to set up background task using a couroutine to not block the UI thread
     */
    private fun delayedInit() {
        applicationScope.launch {
            // Use this instead of the built-in Log
            Timber.plant(Timber.DebugTree())
        }
    }
}